test/integration/smoke/test_vm_life_cycle.py

```diff
@@ -955,6 +956,92 @@ def test_12_start_vm_multiple_volumes_allocated(self):
             "Check virtual machine is in running state"
         )
 
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="false")
+    def test_13_destroy_and_expunge_vm(self):
+        """Test destroy virtual machine with expunge parameter depending on whether the caller's role has expunge permission.
+        """
+        # Setup steps:
+        # 1. Create role with DENY expunge permission.
+        # 2. Create account with said role.
+        # 3. Create a VM of said account.
+        # 4. Create a VM of cls.account
+        # Validation steps:
+        # 1. Destroy the VM with the created account and verify it was not destroyed.
+        # 1. Destroy the other VM with cls.account and verify it was expunged.
+
+        role = Role.importRole(
+            self.apiclient,
+            {
+                "name": "MarvinFake Import Role ",
+                "type": "DomainAdmin",
+                "description": "Fake Import Domain Admin Role created by Marvin test",
+                "rules" : [{"rule":"list*", "permission":"allow","description":"Listing apis"},
+                           {"rule":"get*", "permission":"allow","description":"Get apis"},
+                           {"rule":"update*", "permission":"allow","description":"Update apis"},
+                           {"rule":"queryAsyncJobResult", "permission":"allow","description":"Query async job result"},
+                           {"rule":"deployVirtualMachine", "permission":"allow","description":"Deploy virtual machine"},
+                           {"rule":"destroyVirtualMachine", "permission":"allow","description":"Destroy virtual machine"},
+                           {"rule":"expungeVirtualMachine", "permission":"deny","description":"Expunge virtual machine"}]
+            },
+        )
+
+        domadm = Account.create(
+            self.apiclient,
+            self.services["account"],
+            admin=True,
+            roleid=role.id,
+            domainid=self.domain.id
+        )
+
+        self.cleanup.append(domadm)
+        self.cleanup.append(role)
```

Comment:
```suggestion
        role = Role.importRole(
            self.apiclient,
            {
                "name": "MarvinFake Import Role ",
                "type": "DomainAdmin",
                "description": "Fake Import Domain Admin Role created by Marvin test",
                "rules" : [{"rule":"list*", "permission":"allow","description":"Listing apis"},
                           {"rule":"get*", "permission":"allow","description":"Get apis"},
                           {"rule":"update*", "permission":"allow","description":"Update apis"},
                           {"rule":"queryAsyncJobResult", "permission":"allow","description":"Query async job result"},
                           {"rule":"deployVirtualMachine", "permission":"allow","description":"Deploy virtual machine"},
                           {"rule":"destroyVirtualMachine", "permission":"allow","description":"Destroy virtual machine"},
                           {"rule":"expungeVirtualMachine", "permission":"deny","description":"Expunge virtual machine"}]
            },
        )
        self.cleanup.append(role)

        domadm = Account.create(
            self.apiclient,
            self.services["account"],
            admin=True,
            roleid=role.id,
            domainid=self.domain.id
        )
        self.cleanup.append(domadm)
```

@gpordeus , the reason for this is that any exception in the second create would leave the first object as garbage behind.

Reply:
@DaanHoogland I tried it, but then when it's time to clean up it ends up in error since the role is being deleted before the account. Would adding a manual swap, like this, be better?

```
self.cleanup.append(role)
domadm =[...]
self.cleanup[-1]=domadm
self.cleanup.append(role)
```

Reply:
@DaanHoogland I tried adding the super call, but there are conflicts with test 12 and I don't have the time right now to fix it properly, so I've just added the swap to avoid the possible garbage.

Reply:
ok, can you add a comment about it? fine by me, but people need to be aware and not use this code as an example for how "it should be done".

Reply:
@DaanHoogland Yes, of course, you're right.

I've added the comments and changed the testDeployVM tearDown/tearDownClass since it didn't have any cleanup order issues and it is the first test class one sees in the file.